### PR TITLE
Remove make_obj_array([...]) for scalar DOF arrays

### PR DESCRIPTION
**Pull request by majosm**

### CODE REVIEW

**Comment by inducer** [comment]
FYI @MTCam

**Review by inducer — APPROVED** [review]
LGTM. Good to go once CI passes.

**Comment by inducer** [comment]
(And thanks again for making this happen!)

**Comment by MTCam** [comment]
> FYI @MTCam

Just planning to merge and update prometheus devel after this gets merged.

**Pull request merged**
